src/Acuminator/Acuminator.Analyzers/FixProviders/DAC/ForbiddenFieldsInDacFix.cs

```diff
@@ -0,0 +1,148 @@
+                return document;
+            //Format tabulations
+            AdhocWorkspace workspace = new AdhocWorkspace();
+            OptionSet options = workspace.Options; 
+            options = options.WithChangedOption(FormattingOptions.UseTabs,LanguageNames.CSharp,false)
+                             .WithChangedOption(FormattingOptions.TabSize, LanguageNames.CSharp, 4)
+                             .WithChangedOption(FormattingOptions.IndentationSize, LanguageNames.CSharp, 4);
+            workspace.Options = options;
+            var modifiedRoot = Formatter.Format(root.ReplaceNode(classDeclaration, classModified),workspace,options);
+
+            if (cancellationToken.IsCancellationRequested)
+                return document;
+            return document.WithSyntaxRoot(modifiedRoot);
+        }
+    }
+
+
+    public class RegionClassRewriter : CSharpSyntaxRewriter
+    {
+        //private Stack<RegionDirectiveTriviaSyntax> regionsStack;
+        private Stack<SyntaxTrivia> deletedTriviaStack;
+        private string removableIdentifier;
+        private CancellationToken cancellationToken;
+
+        public RegionClassRewriter(string aRemovableIdentifier,
+                                   CancellationToken cToken)
+                                : base(visitIntoStructuredTrivia: true)
+        {
+            //regionsStack = new Stack<RegionDirectiveTriviaSyntax>();
+            deletedTriviaStack = new Stack<SyntaxTrivia>();
+            removableIdentifier = aRemovableIdentifier.ToUpperInvariant();
+            cancellationToken = cToken;
+        }
+
+        public override SyntaxTrivia VisitTrivia(SyntaxTrivia trivia)
+        {
+            if (deletedTriviaStack == null || cancellationToken.IsCancellationRequested)
+                return base.VisitTrivia(trivia);
+            if (trivia.Kind() == SyntaxKind.RegionDirectiveTrivia)
+            {
+                deletedTriviaStack.Push(trivia);
+                if (trivia.ToString().ToUpperInvariant().Contains(removableIdentifier))
+                {
+                    return SyntaxFactory.Tab;// SyntaxFactory.ElasticEndOfLine(" a") as SyntaxNode;
+                }
+            }
+            if (trivia.Kind()  == SyntaxKind.EndRegionDirectiveTrivia)
+            {
+                if (deletedTriviaStack.Count == 0)
+                    return base.VisitTrivia(trivia);
+                SyntaxTrivia regionStart = deletedTriviaStack.Pop();
+                if (regionStart.ToString().ToUpperInvariant().Contains(removableIdentifier))
+                    return  SyntaxFactory.Tab;
+            }
+            return base.VisitTrivia(trivia);
+        }
+        
+
+        public override SyntaxNode VisitClassDeclaration(ClassDeclarationSyntax node)
+        {
+            if (cancellationToken.IsCancellationRequested)
+                return null;
+            var properties = node.Members.OfType<PropertyDeclarationSyntax>();
```

Comment:
@dropsonic, @saloponov  I feel very doubtful about this particular method. The visitor pattern is not used very good here. You don't need the visitor to just perform this code on a DAC class declaration.
My idea was to use visitor to remove regions from DAC first and then remove all properties/bql field classes since it was problematic to remove both at the same time. Maybe you won't need the formatter at all then